cropper/src/main/java/com/canhub/cropper/CropImage.java

```diff
@@ -272,20 +265,54 @@ public static Intent getCameraIntent(@NonNull Context context, Uri outputFileUri
             allIntents.add(intent);
         }
 
+        // Just in case queryIntentActivities returns emptyList
+        if (allIntents.isEmpty()) {
+            allIntents.add(captureIntent);
+        }
+
         return allIntents;
     }
 
+
     /**
      * Get all Gallery intents for getting image from one of the apps of the device that handle
      * images.
      */
-    public static Intent getGalleryIntent(String action, boolean includeDocuments) {
-        Intent galleryIntent = new Intent();
-        galleryIntent.setAction(action);
+    public static List<Intent> getGalleryIntents(
+            @NonNull PackageManager packageManager, String action, boolean includeDocuments) {
+        List<Intent> intents = new ArrayList<>();
+        Intent galleryIntent = new Intent(action);
         galleryIntent.setType(includeDocuments ? "*/*" : "image/*");
         galleryIntent.addCategory(Intent.CATEGORY_OPENABLE);
 
-        return galleryIntent;
+        List<ResolveInfo> listGallery = packageManager.queryIntentActivities(galleryIntent, 0);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q && listGallery.size() > 2) {
```

Comment:
Please change `Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q` to `CommonVersionCheck.INSTANCE.isAtLeastQ29()`